### DISCUSSION TITLE
Unable to log to OAP from the node.js agent

### DISCUSSION THREAD

**Discussion by PaulCarlucciStratascale** [apache/skywalking#10918] [Q&A] [answered]
I'm using node.js agent 0.6.0 against OAP 9.3.0.  I have an application with the agent enabled and I'm trying to push some logging data at OAP tagged and correlated so for example I could see the userID for a given login transaction.

I've discovered that if I add the following:
`import { ContextManager} from "skywalking-backend-js";`
`ContextManager.currentSpan.log("userid", "12345");`

...then with SW_AGENT_LOGGING_LEVEL = "debug" I can see that it is in fact adding my data into the logs[] array of the current span, however this data never shows up in the OAP UI or the logs or log_tags tables of the PostgreSQL database underlying my OAP instance.

What am I doing wrong here?

**Comment by wu-sheng** [comment]
This is the log of span. You can see from the trace view.
It is not the same thing.
I don't think node agent supports log collecting for now. @kezhenxu94 right?

**Reply by kezhenxu94** [reply, marked as answer]
Yes, node agent doesn't support application log, @PaulCarlucciStratascale what you are doing is trace logs, which is embedded in the trace data, you need to check the trace span detail to see the log

**Reply by PaulCarlucciStratascale** [reply]
Oh, now that I know these are trace logs I see where these show up.  This is really valuable, thank you.  I would have greatly appreciated if it was documented, as it is I had to do a bit of exploring to find that this was even possible.

Is application logging on the horizon for the NodeJS agent?

**Reply by wu-sheng** [reply]
The concept of current span logs is not a new thing actually. So, we don't know have to document this. If we want to write docs, we have to write all concepts in the main doc instead of nodejs doc.
I will see what I can do.

About log collecting, I don't know someone is working on this yet.
